.circleci/config.yml

```diff
@@ -76,20 +84,34 @@ commands:
       tarball-path:
         type: string
         default: solc-js.tgz
-      install-command:
-        type: string
-        default: npm install
+      package-manager:
+        type: enum
+        enum: ["npm", "yarn"]
+        default: npm
     steps:
       - run:
-          name: Inject solc-js from the tarball into dependencies at <<parameters.path>>
+          name: "Sanity check: tarball exists and the target dir contains a JS project"
           command: |
             [[ -f "<<parameters.tarball-path>>" ]]
+            [[ -f "<<parameters.path>>/package.json" ]]
+      - run:
+          name: Inject solc-js from the tarball into dependencies at <<parameters.path>>
+          command: |
             absolute_tarball_path=$(realpath "<<parameters.tarball-path>>")
-            for solc_module in $(find "<<parameters.path>>" -type d -path "*/node_modules/solc"); do
-                pushd "${solc_module}/../.."
-                <<parameters.install-command>> "$absolute_tarball_path" --ignore-workspace-root-check
-                popd
-            done
+            cd "<<parameters.path>>"
+            mv package.json original-package.json
+            # NOTE: The 'overrides' feature requires npm >= 8.3. Yarn requires `resolutions` instead.
+            jq ". + {overrides: {solc: \"${absolute_tarball_path}\"}} + {resolutions: {solc: \"${absolute_tarball_path}\"}}" original-package.json > package.json
+            "<<parameters.package-manager>>" install
```

Comment:
Doing `yarn install` here increases the running time of the Truffle test a lot. That's because it recompiles all TypeScript files and that's really heavy. `install-truffle-dependencies` at this point already ran `yarn install` so we just redo that work.

The problem is that it's hard to avoid it while still keeping it as a separate, self-contained command. I'd have to duplicate this stuff in `install-truffle-dependencies` but still keep it here for Hardhat. Maybe I could figure out something nicer but it's a refactor that I don't really want to go into right now.